homeassistant/components/light/__init__.py

```diff
@@ -305,17 +306,22 @@ class SetIntentHandler(intent.IntentHandler):
                 preprocess_turn_on_alternatives(pars)
                 turn_light_off, off_pars = preprocess_turn_off(pars)
             if turn_light_off:
-                await light.async_turn_off(**off_pars)
+                task = light.async_request_call(light.async_turn_off, **off_pars)
             else:
-                await light.async_turn_on(**pars)
+                task = light.async_request_call(light.async_turn_on, **pars)
 
-            if not light.should_poll:
-                continue
+            change_tasks.append(task)
 
-            update_tasks.append(light.async_update_ha_state(True))
+            if light.should_poll:
+                poll_lights.append(light)
 
-        if update_tasks:
-            await asyncio.wait(update_tasks)
+        if change_tasks:
+            await asyncio.gather(*change_tasks)
```

Comment:
You should not use gather if you not can handle exception. That means he change the state but would never call async_update_ha_state if a integration fail. They messup the complete flow and end up in not stable core. That is very bad for core and HA. At this point I'm not ident with @balloob to mess up the stable execution

Reply:
I changed it back to wait, i assumed that is what you wanted